packages/client/src/pages/Login/Login.tsx

```diff
@@ -1,33 +1,75 @@
-import React from 'react'
+import React, { useState } from 'react'
 import { Layout } from '../../components/Layout'
 import { Card } from '../../components/Card'
-import { Field } from '../../components/UI/Field'
-import { Form } from '../../components/Form'
+import { useForm } from '../../hooks/useForm'
+import { Button } from '../../components/UI/Button'
+import { login } from '../../api/auth'
+import { isAxiosError } from 'axios'
+
+const loginForm = [
+  {
+    name: 'login',
+    type: 'text',
+    label: 'Login',
+    value: '',
+    isValid: true,
+    rule: /^(?!^\d+$)[\dA-Za-z-_]+$/,
+    errorMessage: 'Must consist of letters and numbers, no spaces, no special characters'
+  },
+  {
+    name: 'password',
+    type: 'password',
+    label: 'Password',
+    value: '',
+    isValid: true,
+    rule: /(?=.*\d)(?=.*[A-ZА-ЯЁ])/,
+    errorMessage: 'Must consist of at least one capital letter and number'
+  }
+]
 
 export const Login = () => {
-  const handleSubmit = () => {
-    console.log('summit')
+  const [formValues, isFormValid, formInputs] = useForm(loginForm)
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState('')
+
+  const onSubmit = async (e: React.SyntheticEvent) => {
+    e.preventDefault()
+    const isValid = isFormValid()
+    const formData = formValues()
+    if (isValid) {
```

Comment:
не эконом на пустых строках, каждый логический блок должен быть разделён:

```suggestion
    e.preventDefault()
    
    const isValid = isFormValid()
    const formData = formValues()
    
    if (isValid) {
```

Reply:
done